### PR TITLE
Always use the more portable 3-arg form for open()

### PR DESCRIPTION
**Pull request by hughsie**

### CODE REVIEW

File: plugins/thunderbolt-power/fu-plugin-thunderbolt-power.c [hunk 1]
```diff
@@ -187,7 +188,7 @@ fu_plugin_thunderbolt_power_kernel_force_power (FuPlugin *plugin, gboolean enabl
 		return FALSE;
 	}
 	g_debug ("Setting force power to %d using kernel", enable);
-	fd = g_open (data->force_path, O_WRONLY);
+	fd = g_open (data->force_path, O_WRONLY, S_IWUSR);
```
Comment (gicmo) [inline plugins/thunderbolt-power/fu-plugin-thunderbolt-power.c]: As long as you are not creating the file (`g_creat`, `flags | O_CREAT`), the `mode` gets [ignored](https://github.com/torvalds/linux/blob/9c7db5004280767566e91a33445bf93aa479ef02/fs/open.c#L945), so I think just passing `0` here would be better.

Comment (hughsie) [reply plugins/thunderbolt-power/fu-plugin-thunderbolt-power.c]: Fixed, thanks.

File: src/fu-io-channel.c [hunk 2]
```diff
@@ -16,6 +16,7 @@
 #include <poll.h>
 #endif
 #include <string.h>
+#include <sys/stat.h>
```
Comment (superm1) [inline src/fu-io-channel.c]: Unneeded include, right?

Comment (hughsie) [reply src/fu-io-channel.c]: Nah, actually needed for the S_IRWXU define. I can split out this hunk as another commit if that's better.

**Pull request merged**